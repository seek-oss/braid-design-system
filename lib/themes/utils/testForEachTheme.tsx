import React, { FunctionComponent } from 'react';
import each from 'lodash/each';
import { Omit } from 'utility-types';
import * as themes from '..';
import { Theme } from '../theme';
import ThemeProvider, {
  ThemeProviderProps
} from '../../components/ThemeProvider/ThemeProvider';

type TestCallback = (
  args: {
    theme: Theme;
    ThemeProvider: FunctionComponent<Omit<ThemeProviderProps, 'theme'>>;
  }
) => void;

export default (callback: TestCallback) => {
  each(themes, theme => {
    describe(`Theme: ${theme.name}`, () => {
      callback({
        theme,
        ThemeProvider: props => <ThemeProvider theme={theme} {...props} />
      });
    });
  });
};
