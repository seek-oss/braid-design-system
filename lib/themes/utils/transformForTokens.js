import merge from 'lodash/merge';
import alignTextToGrid from './alignTextToGrid';
import createResponsiveRules from './createResponsiveRules';

export default ({ tokens }) => {
  const rules = [];
  Object.keys(tokens.text).forEach(typeName => {
    const type = tokens.text[typeName];

    const { transform: mobileTransform } = alignTextToGrid(type.mobile, tokens);
    const { transform: desktopTransform } = alignTextToGrid(
      type.desktop,
      tokens
    );

    rules.push(
      createResponsiveRules({
        tokens,
        selector: `.${typeName}Text`,
        mobileRules: {
          transform: mobileTransform
        },
        desktopRules: {
          transform: desktopTransform
        }
      })
    );
  });

  return merge({}, ...rules);
};
