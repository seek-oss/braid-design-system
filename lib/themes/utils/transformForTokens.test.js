import transformForTokens from './transformForTokens';
import testForEachTheme from './testForEachTheme';

describe('transformForTokens', () => {
  testForEachTheme(({ theme }) => {
    test('Rules', () => {
      expect(
        transformForTokens({
          tokens: theme.tokens
        })
      ).toMatchSnapshot();
    });
  });
});
