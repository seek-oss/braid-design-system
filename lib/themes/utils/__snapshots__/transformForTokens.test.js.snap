// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformForTokens Theme: jobStreet Rules 1`] = `
Object {
  ".largeText": Object {
    "transform": "translateY(0.38em)",
  },
  ".standardText": Object {
    "transform": "translateY(0.3442857142857143em)",
  },
}
`;

exports[`transformForTokens Theme: seekAnz Rules 1`] = `
Object {
  ".largeText": Object {
    "transform": "translateY(0.32666666666666666em)",
  },
  ".standardText": Object {
    "transform": "translateY(0.41000000000000003em)",
  },
}
`;

exports[`transformForTokens Theme: seekAsia Rules 1`] = `
Object {
  ".largeText": Object {
    "transform": "translateY(0.29666666666666663em)",
  },
  ".standardText": Object {
    "transform": "translateY(0.38em)",
  },
  "@media screen and (min-width: 768px)": Object {
    ".largeText": Object {
      "transform": "translateY(0.32999999999999996em)",
    },
  },
}
`;

exports[`transformForTokens Theme: wireframe Rules 1`] = `
Object {
  ".largeText": Object {
    "transform": "translateY(0.32666666666666666em)",
  },
  ".standardText": Object {
    "transform": "translateY(0.41000000000000003em)",
  },
}
`;
